crates/services/executor/src/executor.rs

```diff
@@ -1349,7 +1263,35 @@ where
             tx.outputs(),
         )?;
 
-        // We always need to update outputs with storage state after execution.
+        self.update_tx_outputs(tx_st_transaction, tx_id, &mut tx)?;
```

Comment:
I know that it was before like this, but it would be nice if you moved this function to the `attempt_tx_execution_with_vm`. Then we can remove `mut` part from the `tx`=)